app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/CheckYourAnswersController.scala

```diff
@@ -49,15 +49,18 @@ class CheckYourAnswersController @Inject()(override val controllerComponents: Me
   }
 
   val onSubmit: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
+    request.declarationJourney.goodsEntries.declarationGoodsIfComplete
+      .fold(actionProvider.invalidRequestF)(goods => declarationConfirmation(request, goods))
+      .recoverWith {
```

Comment:
Is the recover necessary? I'd imagine the error handler goes us a 500 if the exception is just allowed to propagate?